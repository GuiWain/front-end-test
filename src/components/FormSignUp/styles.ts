import styled from 'styled-components'
import UseLink from 'next/link'

import UseTextField from 'components/TextField'
import UseButton from 'components/Button'

export const Link = styled(UseLink)``
export const TextField = styled(UseTextField)``

export const ButtonSubmit = styled(UseButton).attrs({
  type: 'submit',
  fullWidth: true
})``

export const Wrapper = styled.form`
  width: 35rem;
  display: flex;
  flex-direction: column;
  gap: 1.5rem;

  background-color: #fff;
  box-shadow: 0px 0.3rem 0.6rem #00000029;
  padding: 3rem 4rem;
  margin: auto;
`

export const DivText = styled.div`
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
`

export const DivCheckBox = styled.div`
  display: flex;
  gap: 0.5rem;
  margin: 1rem 0;

  justify-items: center;
  align-items: center;
`

export const Image = styled.img`
  margin: 0 1rem 0 0;
`
