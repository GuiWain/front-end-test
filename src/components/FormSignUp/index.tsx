import * as S from './styles'
import { useState, FormEvent } from 'react'

import { FieldErrors } from 'utils/validations'

export type FormSignUpValue = {
  [field: string]: string
}

export type FormSignUpProps = {
  handleSubmit: (e: FormEvent, values: FormSignUpValue) => void
  fieldError: FieldErrors
}

const FormSignUp = ({ handleSubmit, fieldError }: FormSignUpProps) => {
  const [values, setValues] = useState<FormSignUpValue>({})

  const handleInput = (field: string, value: string) => {
    setValues((s) => ({ ...s, [field]: value }))
  }

  return (
    <S.Wrapper onSubmit={(e) => handleSubmit(e, values)}>
      <S.DivText>
        <S.TextField
          type="text"
          label="Full name"
          placeholder="Your name"
          error={fieldError?.name}
          onInputChange={(s) => handleInput('username', s)}
          name="name"
        />
        <S.TextField
          type="email"
          label="Email"
          placeholder="Your email"
          error={fieldError?.email}
          onInputChange={(s) => handleInput('email', s)}
          name="email"
        />
        <S.TextField
          type="password"
          label="Password"
          error={fieldError?.password}
          onInputChange={(s) => handleInput('password', s)}
          name="passwordSignUp"
        />
        <S.TextField
          type="password"
          label="Confirm password"
          onInputChange={(s) => handleInput('confirm_password', s)}
          name="confirm_password"
          error={fieldError?.confirm_password}
          onPaste={(e) => e.preventDefault()}
          autoComplete="off"
        />
      </S.DivText>
      <S.ButtonSubmit>Sign up</S.ButtonSubmit>
    </S.Wrapper>
  )
}

export default FormSignUp
