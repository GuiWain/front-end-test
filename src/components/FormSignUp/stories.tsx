import { Story, Meta } from '@storybook/react/types-6-0'
import FormSignUp, { FormSignUpProps } from '.'

import { FormEvent } from 'react'

export default {
  title: 'Form/FormSignUp',
  component: FormSignUp
} as Meta

const handleSubmit = (e: FormEvent) => {
  e.preventDefault()
  return
}

export const Default: Story<FormSignUpProps> = (args) => (
  <FormSignUp {...args} />
)

Default.args = {
  handleSubmit: handleSubmit,
  fieldError: { name: '', email: '', password: '', confirm_password: '' }
}
