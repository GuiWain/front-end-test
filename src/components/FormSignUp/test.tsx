import { fireEvent, render, screen } from 'utils/test-utils'
import userEvent from '@testing-library/user-event'
import FormSignUp from '.'

describe('<FormSignUp />', () => {
  it('should render the FormSignUp', () => {
    const handleSubmit = jest.fn()
    const fieldError = {}
    const { container } = render(
      <FormSignUp handleSubmit={handleSubmit} fieldError={fieldError} />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be prevent paste text in confirm password', () => {
    const handleSubmit = jest.fn()
    const fieldError = {}
    render(<FormSignUp handleSubmit={handleSubmit} fieldError={fieldError} />)

    const inputPasswordConfirm = screen.getByLabelText(/Confirm password/i)
    const passwordConfirm = 'password12345'
    userEvent.paste(inputPasswordConfirm, passwordConfirm)

    expect(inputPasswordConfirm).toHaveValue()
  })

  it('should be submit form with sucess', () => {
    const handleSubmit = jest.fn()
    const fieldError = {}
    render(<FormSignUp handleSubmit={handleSubmit} fieldError={fieldError} />)

    const inputName = screen.getByPlaceholderText(/Your name/i)
    const name = 'Name name name'
    userEvent.type(inputName, name)

    const inputEmail = screen.getByPlaceholderText(/Your email/i)
    const email = 'success@email.com.br'
    userEvent.type(inputEmail, email)

    const inputPassword = screen.getByLabelText('Password')
    const password = 'success123456'
    userEvent.type(inputPassword, password)

    const inputPasswordConfirm = screen.getByLabelText(/Confirm password/i)
    const passwordConfirm = 'success123456'
    userEvent.type(inputPasswordConfirm, passwordConfirm)

    const submit = screen.getByText('Sign up')
    fireEvent.submit(submit)
    expect(handleSubmit).toBeCalled()
  })
})
