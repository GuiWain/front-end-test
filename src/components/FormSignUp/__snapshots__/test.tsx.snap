// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignUp /> should render the FormSignUp 1`] = `
.c6 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c7 {
  cursor: pointer;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF 0% 0% no-repeat padding-box;
  border: 0.1rem solid #CCC8C8;
  border-radius: 0.4rem;
  padding: 0 0.8rem;
  height: 3.6rem;
}

.c3:focus-within {
  box-shadow: 0rem 0.4rem 0.6rem #00000014;
  border: 0.1rem solid #3241A1;
}

.c4 {
  color: #4F4C4C;
  font: normal normal 400 1.4rem / 2.8rem 'Open Sans','Roboto','Helvetica Neue',sans-serif;
  padding: 0.8rem 0;
  padding-left: 0.4rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c4:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 2.4 #EAEAEA inset;
  -webkit-filter: none;
  filter: none;
}

.c8 {
  color: #4F4C4C;
  font: normal normal 400 1.4rem / 2.8rem 'Open Sans','Roboto','Helvetica Neue',sans-serif;
  padding: 0.8rem 0;
  padding-right: 0.4rem;
  background: transparent;
  border: 0;
  outline: none;
  width: calc(100% - 2.2rem);
}

.c8:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 2.4 #EAEAEA inset;
  -webkit-filter: none;
  filter: none;
}

.c2 {
  font: normal normal 600 1.6rem / 4rem 'Open Sans','Roboto','Helvetica Neue',sans-serif;
  -webkit-letter-spacing: -0.006rem;
  -moz-letter-spacing: -0.006rem;
  -ms-letter-spacing: -0.006rem;
  letter-spacing: -0.006rem;
  text-align: left;
  color: #4F4C4C;
  cursor: pointer;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #848080;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c5 > svg {
  width: 2rem;
  height: 100%;
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.6rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border-radius: 0.4rem;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  color: #FFFFFF;
  font-family: 'Open Sans','Roboto','Helvetica Neue',sans-serif;
  font-weight: 700;
  font-size: 1.4rem;
  line-height: 2.6rem;
  width: 100%;
  height: 3.6rem;
  padding: 0.9rem 4.4rem;
  font-size: 1.4rem;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  background: #510161 0% 0% no-repeat padding-box;
  box-shadow: 0rem 0.4rem 0.8rem #00000014;
  border: none;
}

.c9:hover {
  background: #8802A3 0% 0% no-repeat padding-box;
  -webkit-transition: background 0.5s ease-in-out;
  transition: background 0.5s ease-in-out;
}

.c0 {
  width: 35rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.5rem;
  background-color: #fff;
  box-shadow: 0px 0.3rem 0.6rem #00000029;
  padding: 3rem 4rem;
  margin: auto;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.5rem;
}

<form
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class=""
    >
      <label
        class="c2"
        for="name"
      >
        Full name
      </label>
      <div
        class="c3"
      >
        <input
          class="c4 "
          id="name"
          name="name"
          placeholder="Your name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class=""
    >
      <label
        class="c2"
        for="email"
      >
        Email
      </label>
      <div
        class="c3"
      >
        <input
          class="c4 "
          id="email"
          name="email"
          placeholder="Your email"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      class=""
    >
      <label
        class="c2"
        for="passwordSignUp"
      >
        Password
      </label>
      <div
        class="c3"
      >
        <div
          class="c5"
        >
          <svg
            aria-hidden="true"
            class="c6 c7"
            data-testid="eye-slash-icon"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.359 11.238C15.06 9.72 16 8 16 8s-3-5.5-8-5.5a7.028 7.028 0 0 0-2.79.588l.77.771A5.944 5.944 0 0 1 8 3.5c2.12 0 3.879 1.168 5.168 2.457A13.134 13.134 0 0 1 14.828 8c-.058.087-.122.183-.195.288-.335.48-.83 1.12-1.465 1.755-.165.165-.337.328-.517.486l.708.709z"
            />
            <path
              d="M11.297 9.176a3.5 3.5 0 0 0-4.474-4.474l.823.823a2.5 2.5 0 0 1 2.829 2.829l.822.822zm-2.943 1.299.822.822a3.5 3.5 0 0 1-4.474-4.474l.823.823a2.5 2.5 0 0 0 2.829 2.829z"
            />
            <path
              d="M3.35 5.47c-.18.16-.353.322-.518.487A13.134 13.134 0 0 0 1.172 8l.195.288c.335.48.83 1.12 1.465 1.755C4.121 11.332 5.881 12.5 8 12.5c.716 0 1.39-.133 2.02-.36l.77.772A7.029 7.029 0 0 1 8 13.5C3 13.5 0 8 0 8s.939-1.721 2.641-3.238l.708.709zm10.296 8.884-12-12 .708-.708 12 12-.708.708z"
            />
          </svg>
        </div>
        <input
          class="c8 "
          id="passwordSignUp"
          name="passwordSignUp"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class=""
    >
      <label
        class="c2"
        for="confirm_password"
      >
        Confirm password
      </label>
      <div
        class="c3"
      >
        <div
          class="c5"
        >
          <svg
            aria-hidden="true"
            class="c6 c7"
            data-testid="eye-slash-icon"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.359 11.238C15.06 9.72 16 8 16 8s-3-5.5-8-5.5a7.028 7.028 0 0 0-2.79.588l.77.771A5.944 5.944 0 0 1 8 3.5c2.12 0 3.879 1.168 5.168 2.457A13.134 13.134 0 0 1 14.828 8c-.058.087-.122.183-.195.288-.335.48-.83 1.12-1.465 1.755-.165.165-.337.328-.517.486l.708.709z"
            />
            <path
              d="M11.297 9.176a3.5 3.5 0 0 0-4.474-4.474l.823.823a2.5 2.5 0 0 1 2.829 2.829l.822.822zm-2.943 1.299.822.822a3.5 3.5 0 0 1-4.474-4.474l.823.823a2.5 2.5 0 0 0 2.829 2.829z"
            />
            <path
              d="M3.35 5.47c-.18.16-.353.322-.518.487A13.134 13.134 0 0 0 1.172 8l.195.288c.335.48.83 1.12 1.465 1.755C4.121 11.332 5.881 12.5 8 12.5c.716 0 1.39-.133 2.02-.36l.77.772A7.029 7.029 0 0 1 8 13.5C3 13.5 0 8 0 8s.939-1.721 2.641-3.238l.708.709zm10.296 8.884-12-12 .708-.708 12 12-.708.708z"
            />
          </svg>
        </div>
        <input
          autocomplete="off"
          class="c8 "
          id="confirm_password"
          name="confirm_password"
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
  <button
    class="c9 "
    type="submit"
  >
    Sign up
  </button>
</form>
`;
